core/src/Streamly/Internal/FileSystem/Posix/ReadDir.hsc

```diff
@@ -163,6 +180,10 @@ readDirStreamEither (DirStream dirp) = loop
         -- fromPtrN, but it is not straightforward because the reclen is
         -- padded to 8-byte boundary.
         name <- Array.fromCString (castPtr dname)
+
+        when (dtype == #const DT_UNKNOWN)
+            $ throwIO (RDE_DT_UNKNOWN (mkPath name))
```

Comment:
We cannot throw an exception, we need to use stat to find out the file type in this case. It will be less efficient but it will work. We cannot fail based on filesystem features.